### PR TITLE
Explicitly set InputItem.defaultProps type to avoid type collision

### PR DESCRIPTION
**Pull request by gregwym**
The generated type for `InputItem.defaultProps.type` (string) is not the same as `InputItemPropsType.type` (union of string literals). This causes problem when using antd together with `styled-components`. 

image

Similar practice can be found in `ListItem` and `AntTabBar`.

---
This change is [Reviewable](https://reviewable.io/reviews/ant-design/ant-design-mobile/2785)

### CODE REVIEW

**Comment by gregwym** [comment]
Will anybody review this?

**Comment by doxiaodong** [comment]
How about other components with defaultProps?

**Comment by gregwym** [comment]
@doxiaodong I was trying to keep the change to a minimum. 
This issue only exists for props with string enum values in them. Please see my latest commit where I updated all instances that are likely to have this issue. I have also deleted the default prop values that are no longer exist in the type definition or used in the component logic.

File: components/popover/index.tsx [hunk 1]
```diff
@@ -43,11 +43,9 @@ function recursiveCloneChildren(
 }
 
 export default class Popover extends React.Component<PopOverPropsType, any> {
-  static defaultProps = {
+  static defaultProps: Partial<PopOverPropsType> = {
     prefixCls: 'am-popover',
     placement: 'bottomRight',
-    align: { overflow: { adjustY: 0, adjustX: 0 } },
```
Comment (doxiaodong) [inline components/popover/index.tsx]: Add `align` and `trigger` to PopOverPropsType instead of removing it.

File: components/result/index.tsx [hunk 2]
```diff
@@ -11,9 +11,8 @@ export interface ResultProps extends ResultPropsType {
 }
 
 export default class Result extends React.Component<ResultProps, any> {
-  static defaultProps = {
+  static defaultProps: Partial<ResultProps> = {
     prefixCls: 'am-result',
-    buttonType: '',
```
Comment (doxiaodong) [inline components/result/index.tsx]: ok, button 本身没有定义 '' type 类型，这里可以直接不设置默认值

File: components/notice-bar/index.tsx [hunk 3]
```diff
@@ -12,9 +12,8 @@ export interface NoticeWebProps extends NoticeBarPropsType {
 }
 
 export default class NoticeBar extends React.Component<NoticeWebProps, any> {
-  static defaultProps = {
+  static defaultProps: Partial<NoticeWebProps> = {
     prefixCls: 'am-notice-bar',
-    mode: '',
```
Comment (doxiaodong) [inline components/notice-bar/index.tsx]: mode: 'closable' | 'link' | undefined , 所以这里也可以直接移除掉

File: components/modal/Modal.tsx [hunk 4]
```diff
@@ -40,14 +40,13 @@ export interface ModalProps extends ModalPropsType<React.CSSProperties> {
...
 export default class Modal extends ModalComponent<ModalProps, any> {
-  static defaultProps = {
+  static defaultProps: Partial<ModalProps> = {
     prefixCls: 'am-modal',
     transparent: false,
     popup: false,
     animationType: 'slide-down',
     animated: true,
     style: {},
-    onShow() {},
```
Comment (doxiaodong) [inline components/modal/Modal.tsx]: `onShow` cannot be removed, add it on `ModalProps`

File: components/date-picker-view/date-picker-view.tsx [hunk 5]
```diff
@@ -8,13 +8,12 @@ export default class DatePickerView extends React.Component<
...
   any
 > {
-  static defaultProps = {
+  static defaultProps: Partial<DatePickerProps> = {
     mode: 'datetime',
     extra: '请选择',
     prefixCls: 'am-picker',
     pickerPrefixCls: 'am-picker-col',
     minuteStep: 1,
-    use12Hours: false,
```
Comment (doxiaodong) [inline components/date-picker-view/date-picker-view.tsx]: `use12Hours` can not be removed, add it on `DatePickerProps`

File: components/carousel/index.tsx [hunk 6]
```diff
@@ -22,13 +22,11 @@ export default class Carousel extends React.Component<
...
   CarouselState
 > {
-  static defaultProps = {
+  static defaultProps: Partial<CarouselProps> = {
     prefixCls: 'am-carousel',
     dots: true,
-    arrows: false,
     autoplay: false,
     infinite: false,
-    cellAlign: 'center',
```
Comment (doxiaodong) [inline components/carousel/index.tsx]: Add `cellAlign` and `arrows` to CarouselProps
```
  arrows?: boolean;
  cellAlign?: 'left' | 'center' | 'right';
```

File: components/activity-indicator/index.tsx [hunk 7]
```diff
@@ -12,11 +12,10 @@ export default class ActivityIndicator extends React.Component<
   ActivityIndicatorProps,
   any
 > {
-  static defaultProps = {
+  static defaultProps: Partial<ActivityIndicatorProps> = {
     prefixCls: 'am-activity-indicator',
     animating: true,
     size: 'small',
-    panelColor: 'rgba(34,34,34,0.6)',
```
Comment (doxiaodong) [inline components/activity-indicator/index.tsx]: ok, `panelColor` can be removed.

**Comment by doxiaodong** [comment]
No response. Related https://github.com/ant-design/ant-design-mobile/issues/3022

**Comment by gregwym** [comment]
Sorry, I couldn’t find spare for this. Will try to work on it later this week.

**Pull request closed without merging**